lib/improver/threshold.py

```diff
@@ -173,6 +188,16 @@ def process(self, input_cube):
         if np.isnan(input_cube.data).any():
             raise ValueError("Error: NaN detected in input cube data")
 
+        # if necessary, convert thresholds and fuzzy bounds into cube units
```

Comment:
I think this process method could do with more explanation regarding how the fuzzy factor will be used, and what it represents. I think a simple example in the docstring would be useful. An example might be:
`
For a 6 mm/hr threshold with a 0.75 fuzzy factor, a range between (6*0.75=) 4.5 and (6*(2-0.75)=) 7.5 would be generated. The thresholded values within this range are scaled so that 4.5 mm/hr yields a thresholded value of 0 and 7.5 mm/hr yields a thresholded value of 1. Therefore, in this case, the fractional values between 4.5 mm/hr and 7.5 mm/hr would follow the pattern:
`

Fractional value | Data value
------------------|------------
0 | 4.5
0.167 | 5.0
0.333 | 5.5
0.5 | 6.0
0.666 | 6.5
0.833 | 7.0
1.0 | 7.5

Reply:
Agreed, although I think this explanation would be better placed in the plugin initialisation, where fuzzy bounds are calculated from the fuzzy factors.  The docstring for process should probably have an explanation of how / why the unit conversion is applied where it is, which I'll add.